libraries/Microsoft.Bot.Builder.Prompts/ConfirmPrompt.cs

```diff
@@ -36,17 +39,100 @@ public string Text
     /// <summary>
     /// ConfirmPrompt recognizes confrimation expressions as bool 
     /// </summary>
-    public class ConfirmPrompt : BasePrompt<ConfirmResult>
+    public class ConfirmPrompt
     {
         private readonly IModel model;
+        public ListStyle Style { get; set; }
+        public PromptValidator<ConfirmResult> Validator { get; set; }
+        public string Culture { get; set; }
+        public List<string> Choices { get; set; }
+        public ChoiceFactoryOptions ChoiceOptions { get; set; }
 
-        public ConfirmPrompt(string culture, PromptValidator<ConfirmResult> validator = null)
-            : base(validator)
+        private static readonly Dictionary<string, List<string>> ConfirmOptions = new Dictionary<string, List<string>>()
+        {
+            { Spanish, new List<string>{ "Sí", "No"} },
+            { Dutch, new List<string>{ "Ja", "Niet"} },
+            { English, new List<string>{ "Yes", "No"} },
+            { French, new List<string>{ "Oui", "Non"} },
+            { German, new List<string>{ "Ja", "Nein"} },
+            { Japanese, new List<string>{ "はい", "いいえ" } },
+            { Portuguese, new List<string>{ "Sim", "Não" } },
+            { Chinese, new List<string>{ "是的", "不" } }
+        };
```

Comment:
The other thing is I would make these public in a way that a developer can customize them if they wish.

---
In reply to: [190346586](https://github.com/Microsoft/botbuilder-dotnet/pull/603#discussion_r190346586) [](ancestors = 190346586)

Reply:
I refactor this, and the inline connectors.
I created a public property which can be passed in the constructor or set anytime. It has a private default value, used if you don't set this property.
Nevertheless, It's important to note how this will affect to validation/recognition of selection, as this is done with a fixed (one for each language) Regex at the moment.